curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/steamroller.english.md

```diff
@@ -33,7 +33,7 @@ tests:
   - text: <code>steamrollArray([1, {}, [3, [[4]]]])</code> should return <code>[1, {}, 3, 4]</code>.
     testString: assert.deepEqual(steamrollArray([1, {}, [3, [[4]]]]), [1, {}, 3, 4]);
   - text: Your solution should not use the <code>Array.prototype.flat()</code> or <code>Array.prototype.flatMap()</code> methods.
-    testString: assert(!code.match(/\.?[\s\S]*?flat/) && !code.match(/\.?[\s\S]*?flatMap/));
+    testString: assert(!code.match(/\w+?\s*\.flat\(\)/) && !code.match(/\w+?\s*\.flatMap\(\)/));
```

Comment:
```suggestion
    testString: assert(!/\.\s*flat(?:Map)?\W|\[\s*(["'`])flat(?:Map)?\1\s*\]/.test(code));
```

Will catch `.flat`, `['flat']` as well as `.flatMap` and `['flatMap']` using any combination of literal strings